### PR TITLE
cmd/charmd: update the command to work more effectively under systemd

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2186/

**Review by fabricematrat — APPROVED** [review]

File: cmd/charmd/main.go [hunk 1]
```diff
@@ -40,25 +40,32 @@ func main() {
...
 	logger.Infof("reading configuration")
-	conf, err := config.Read(confPath)
+	conf, err := config.Read(flag.Arg(0))
 	if err != nil {
-		return errgo.Notef(err, "cannot read config file %q", confPath)
+		fmt.Fprintf(os.Stderr, "STOP: %s\n", err)
+		os.Exit(2)
 	}
+	if *loggingConfig == "" && conf.LoggingConfig != "" {
+		if err := loggo.ConfigureLoggers(*loggingConfig); err != nil {
```
Comment (rogpeppe) [inline cmd/charmd/main.go]: This looks wrong to me. I think you want conf.LoggingConfig here, otherwise this line is guaranteed to do nothing.

**Review by rogpeppe — APPROVED** [review]
LGTM modulo the logging config issue, thanks.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2187/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2188/

**Pull request merged**